Mathlib/Order/Bounds/Basic.lean

```diff
@@ -906,10 +906,20 @@ theorem IsGLB.exists_between' (h : IsGLB s a) (h' : a ∉ s) (hb : a < b) : ∃
 
 end LinearOrder
 
-theorem isGreatest_himp [GeneralizedHeytingAlgebra α] (a b : α) :
+theorem isGreatest_himp [GeneralizedHeytingAlgebra α] {a b : α} :
     IsGreatest {w | w ⊓ a ≤ b} (a ⇨ b) := by
   simp [IsGreatest, mem_upperBounds]
 
-theorem isLeast_sdiff [GeneralizedCoheytingAlgebra α] (a b : α) :
+theorem isLeast_sdiff [GeneralizedCoheytingAlgebra α] {a b : α} :
     IsLeast {w | a ≤ b ⊔ w} (a \ b) := by
   simp [IsLeast, mem_lowerBounds]
+
+theorem isGreatest_compl [HeytingAlgebra α] {a : α} :
+    IsGreatest {w | Disjoint w a} (aᶜ) := by
+  simp only [← himp_bot,disjoint_iff_inf_le]
+  exact isGreatest_himp
+
+theorem isLeast_hnot [CoheytingAlgebra α] {a : α} :
+    IsLeast {w | Codisjoint a w} (￢a) := by
+  simp only [← CoheytingAlgebra.top_sdiff, codisjoint_iff_le_sup]
+  exact isLeast_sdiff
```

Comment:
Here is a golf
```suggestion
theorem isGreatest_compl [HeytingAlgebra α] {a : α} : IsGreatest {w | Disjoint w a} aᶜ := by
  simp [IsGreatest, mem_upperBounds, le_compl_iff_disjoint_right, disjoint_compl_left]

theorem isLeast_hnot [CoheytingAlgebra α] {a : α} : IsLeast {w | Codisjoint a w} (￢a) := by
  simp [IsLeast, mem_lowerBounds, hnot_le_iff_codisjoint_right, codisjoint_hnot_right]
```

Reply:
I think my golf above is better?

Reply:
I didn't see it before writing mine. My proof has the advantage of being independent of the `sdiff`/`himp` results

Reply:
I think that's only an advantage if there is a scenario in which they hold more generally.